### PR TITLE
 Add user friendly error when Java < 17 is used (Issue #280)

### PR DESCRIPTION
**Pull request by AnkushThakur27**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -127,6 +127,38 @@
...
+                                </goals>
+                                <configuration>
+                                    <rules>
+                                        <requireJavaVersion>
+                                            <message>
+                                                🚫 Java version error: This project requires Java ${java.version} or higher.
+
+                                                Your current Java version is ${java.version.detected}
+                                                To fix this:
+                                                1. Download and install Java 17 or higher from your preferred vendor
```
Comment (arey) [inline pom.xml:147]: We could use hère the java.version property

File: pom.xml [hunk 2]
```diff
@@ -127,6 +127,38 @@
...
+                                                2. Make sure JAVA_HOME points to the new Java installation
+                                                3. Verify with 'java -version' command
+                                                4. Run the build again
+
+                                                For more information, check the project documentation.
+                                            </message>
+                                            <version>[${java.version},)</version>
```
Comment (arey) [inline pom.xml:154]: Are you sure of this syntaxe?

Comment (AnkushThakur27) [reply pom.xml:154]: i will take look at night

File: pom.xml (hunk 1 above)
Comment (AnkushThakur27) [reply pom.xml:147]: sure

**Comment by arey** [comment]
Everything looks good. Thank you for your first contribution @AnkushThakur27 AnkushThakur27

File: pom.xml (hunk 2 above)
Comment (arey) [reply pom.xml]: The github action build has failed:
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.4.1:enforce (enforce-java) on project spring-petclinic-admin-server: 
Error:  Rule 0: org.apache.maven.enforcer.rules.version.RequireJavaVersion failed with message:
Error:  The requested JDK version [17) is invalid. Single version must be surrounded by []: [17)
Error:  -> [Help 1]
```

Comment (AnkushThakur27) [reply pom.xml]: @arey let me check that

Comment (AnkushThakur27) [reply pom.xml]: @arey Fixed and pushed just now

**Comment by arey** [comment]
Thank you @AnkushThakur27  👍

**Comment by AnkushThakur27** [comment]
No problem let me know if you want me to work on any requirement from
scratch.


Thanks
Ankush






--
Best Regards,
Ankush Thakur
***@***.***
Mobile:+91 9421057141

**Pull request merged**